### PR TITLE
Merge osm-editor branch into new-search branch.

### PR DESCRIPTION
**Pull request by deathbaba**
This is a copy of upstream/osm-editor branch, to merge into new-search branch. All commits were already reviewed and merged into osm-editor.

@vng @ygorshenin @mpimenov @yunikkk please quick-check these changes and leave your LGTM or any comments. Let's merge it today.

@yunikkk you will need to rebase your local android branch to new-search after this request is merged.

### CODE REVIEW

File: android/jni/com/mapswithme/maps/editor/OpeningHours.cpp [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+jfieldID g_fidEnd;
+// ID-s for Timetable class
+jclass g_clazzTimetable;
+jmethodID g_ctorTimetable;
+jfieldID g_fidWorkingTimespan;
+jfieldID g_fidClosedTimespans;
+jfieldID g_fidIsFullday;
+jfieldID g_fidWeekdays;
+
+jobject JavaHoursMinutes(JNIEnv * env, jlong hours, jlong minutes)
```
Comment (ygorshenin) [inline android/jni/com/mapswithme/maps/editor/OpeningHours.cpp]: How about to return jobject by unique_ptr with custom deleter?

Comment (deathbaba) [inline android/jni/com/mapswithme/maps/editor/OpeningHours.cpp]: @yunikk works on it in a separate pull request now.

File: base/base_tests/range_iterator_test.cpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  {
+    vector<int> result;
+    for (auto const i : ReverseRange(5))
+      result.push_back(i);
+    TEST_EQUAL(result, (vector<int>{4, 3, 2, 1, 0}), ());
+  }
+
+  {
+    vector<int> result;
+    for (auto const i : ReverseRange(2, 5))
```
Comment (ygorshenin) [inline base/base_tests/range_iterator_test.cpp]: IMHO the naming is confusing, especially if we're going to compare ReverseRange with haskell list generators or lisp loops. How about to rename ReverseRange to DownTo(from, to), where DownTo generates a list [from, from - 1, ..., to], not [to, to - 1, ..., from]. Also, Range can be renamed to UpTo(). @deathbaba @mgsergio WDYT?

File: base/collection_cast.hpp [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+namespace details
+{
+template <typename T>
+struct ValueType
+{
+  using type = typename std::remove_reference<T>::type::value_type;
+};
+
+template <typename T>
+using ValueTypeT = typename ValueType<T>::type;
```
Comment (ygorshenin) [inline base/collection_cast.hpp]: TValueType

File: base/collection_cast.hpp [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+#include <type_traits>
+
+namespace my
+{
+namespace details
+{
+template <typename T>
+struct ValueType
+{
+  using type = typename std::remove_reference<T>::type::value_type;
```
Comment (ygorshenin) [inline base/collection_cast.hpp]: TType

File: base/collection_cast.hpp [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+template <typename T>
+using ValueTypeT = typename ValueType<T>::type;
+}  // namespace details
+
+// Use this function to cast one collection to annother.
+// I.E. list<int> const myList = collection_cast<list>(vector<int>{1, 2, 4, 5});
+// More examples:
+// auto const mySet = collection_cast<set>("aaabcccd");
+// auto const myMap = collection_cast<map>(vector<pair<int, int>>{{1, 2}, {3, 4}});
+template <template<typename ... Args> class To, typename From>
```
Comment (ygorshenin) [inline base/collection_cast.hpp]: TArgs, TTo, TFrom :(

File: base/logging.hpp [hunk 6]
```diff
@@ -25,6 +25,19 @@ namespace my
...
   void LogMessageDefault(LogLevel level, SrcPoint const & srcPoint, string const & msg);
   void LogMessageTests(LogLevel level, SrcPoint const & srcPoint, string const & msg);
+
+  /// Scope Guard to temporarily suppress specific log level, for example, in unit tests:
+  /// ...
+  /// {
+  ///   LogLevelSuppressor onlyLERRORAndLCriticalLogsAreEnabled;
+  ///   TEST(SomeFunctionWhichHasDebugOrInfoOrWarningLogs(), ());
+  /// }
+  struct LogLevelSuppressor
```
Comment (ygorshenin) [inline base/logging.hpp]: s/LogLevelSuppressor/ScopedLogLevel/ ?

File: editor/osm_auth.cpp [hunk 7]
```diff
@@ -0,0 +1,418 @@
...
+constexpr char const * kGoogleOAuthPart = "/auth/google?referer=%2Foauth%2Fauthorize%3Foauth_token%3D";
+
+namespace
+{
+inline bool IsKeySecretValid(TKeySecret const & t)
+{
+  return !(t.first.empty() || t.second.empty());
+}
+
+string findAuthenticityToken(string const & body)
```
Comment (ygorshenin) [inline editor/osm_auth.cpp]: FindAutheniticityToken

File: editor/osm_auth.cpp [hunk 8]
```diff
@@ -0,0 +1,418 @@
...
+  string const kValue = "value=\"";
+  auto start = body.find(kValue, pos);
+  if (start == string::npos)
+    return string();
+  start += kValue.length();
+  auto const end = body.find("\"", start);
+  return end == string::npos ? string() : body.substr(start, end - start);
+}
+
+string buildPostRequest(map<string, string> const & params)
```
Comment (ygorshenin) [inline editor/osm_auth.cpp]: BuildPostRequest

File: editor/osm_auth.cpp [hunk 9]
```diff
@@ -0,0 +1,418 @@
...
+  {
+    if (it != params.begin())
+      result += "&";
+    result += it->first + "=" + UrlEncode(it->second);
+  }
+  return result;
+}
+
+// Trying to determine whether it's a login page.
+bool isLoggedIn(string const & contents)
```
Comment (ygorshenin) [inline editor/osm_auth.cpp]: IsLoggedIn

File: base/logging.hpp (hunk 6 above)
Comment (deathbaba) [inline base/logging.hpp]: Both names have pros and cons. ScopedLogLevel does not tell much about it's use case: temporarily change good default behavior.

File: editor/osm_auth.cpp [hunk 10]
```diff
@@ -0,0 +1,418 @@
...
+  if (!request.RunHTTPRequest())
+    return string();
+
+  string const callbackURL = request.url_received();
+  string const vKey = "oauth_verifier=";
+  auto const pos = callbackURL.find(vKey);
+  if (pos == string::npos)
+    return string();
+  auto const end = callbackURL.find("&", pos);
+  return callbackURL.substr(pos + vKey.length(), end == string::npos ? end : end - pos - vKey.length()+ 1);
```
Comment (ygorshenin) [inline editor/osm_auth.cpp]: Why do we need +1 here?
BTW, space before + please.

File: editor/osm_auth.cpp [hunk 11]
```diff
@@ -0,0 +1,418 @@
...
+    return AuthResult::NetworkError;
+  if (request.error_code() != 200)
+    return AuthResult::ServerError;
+  if (!request.was_redirected())
+    return AuthResult::FailLogin;
+  return isLoggedIn(request.server_response()) ? AuthResult::OK : AuthResult::FailLogin;
+}
+
+// Fakes a buttons press, so a user accepts requested permissions.
+string OsmOAuth::SendAuthRequest(string const & requestTokenKey, SessionID const & sid) const
```
Comment (ygorshenin) [inline editor/osm_auth.cpp]: Could you please add tests for this function?

**Comment by mpimenov** [comment]
search/reverse_geocoder.cpp:
В GetNearbyFeatureStreets

Мы пишем
```void doit(params, vector<> & result)```
вместо
```vector<> doit(params)
```

File: editor/opening_hours_ui.hpp [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+#include "std/vector.hpp"
+
+#include "3party/opening_hours/opening_hours.hpp"
+
+namespace editor
+{
+namespace ui
+{
+
+using TOpeningDays = set<osmoh::Weekday>;
```
Comment (ygorshenin) [inline editor/opening_hours_ui.hpp]: Please, do not use using declarations at namespace levels in headers - they may lead to ODR violations and, as consequence, undefined behaviour.

File: editor/osm_auth.cpp [hunk 13]
```diff
@@ -0,0 +1,418 @@
...
+#include "private.h"
+
+#include "3party/liboauthcpp/include/liboauthcpp/liboauthcpp.h"
+#include "3party/Alohalytics/src/http_client.h"
+
+using alohalytics::HTTPClientPlatformWrapper;
+
+namespace osm
+{
+constexpr char const * kApiVersion = "/api/0.6";
```
Comment (ygorshenin) [inline editor/osm_auth.cpp]: constexpr is useless here.

File: editor/osm_auth.hpp [hunk 14]
```diff
@@ -0,0 +1,122 @@
+#pragma once
+
+#include "std/string.hpp"
+#include "std/utility.hpp"
+
+namespace osm
+{
+
+using TKeySecret = pair<string /*key*/, string /*secret*/>;
```
Comment (ygorshenin) [inline editor/osm_auth.hpp]: Please, do not use using declarations at namespace level in headers. Move them to the corresponding class scope.

File: editor/osm_auth.cpp (hunk 11 above)
Comment (Zverik) [inline editor/osm_auth.cpp]: The whole workflow including this function is tested in `osm_auth_tests.cpp`. The function itself is private.

File: editor/server_api.cpp [hunk 15]
```diff
@@ -0,0 +1,125 @@
...
+
+  ostringstream stream;
+  stream << "<osm>\n"
+  "<changeset>\n";
+  for (auto const & tag : kvTags)
+    stream << "  <tag k=\"" << tag.first << "\" v=\"" << tag.second << "\"/>\n";
+  stream << "</changeset>\n"
+  "</osm>\n";
+
+  OsmOAuth::Response const response = m_auth.Request("/changeset/create", "PUT", move(stream.str()));
```
Comment (ygorshenin) [inline editor/server_api.cpp]: No need in move, stream.str() is already rvalue.

File: editor/server_api.cpp [hunk 16]
```diff
@@ -0,0 +1,125 @@
...
+  }
+  else
+    LOG(LWARNING, ("CreateChangeSet request has failed:", response.first));
+
+  return false;
+}
+
+bool ServerApi06::CreateNode(string const & nodeXml, uint64_t & outCreatedNodeId) const
+{
+  OsmOAuth::Response const response = m_auth.Request("/node/create", "PUT", move(nodeXml));
```
Comment (ygorshenin) [inline editor/server_api.cpp]: move is useless here, as nodeXml is const reference. One can not simple steal internals from const references.

File: base/logging.hpp (hunk 6 above)
Comment (ygorshenin) [inline base/logging.hpp]: So, ScopedLogLevelChanger looks nice - it's name describes that it's scoped and that it changes default log level.

File: indexer/osm_editor.cpp [hunk 17]
```diff
@@ -0,0 +1,622 @@
...
+      fields(fields),
+      name(name),
+      address(address)
+  {
+  }
+
+  TEditableFields const fields;
+  bool const name;
+  // Address == true implies Street, House Number, Phone, Fax, Opening Hours, Website, EMail, Postcode.
+  bool const address;
```
Comment (mpimenov) [inline indexer/osm_editor.cpp]: m_fields, m_name, m_address

File: base/logging.hpp (hunk 6 above)
Comment (ygorshenin) [inline base/logging.hpp]: BTW, this class is racy, as it manipulates with global variable without any synchronization. Please, fix it.

File: editor/osm_auth.cpp (hunk 11 above)
Comment (ygorshenin) [inline editor/osm_auth.cpp]: So what? What's the problem to test private function?

Comment (ygorshenin) [inline editor/osm_auth.cpp]: I'm asking for a tests for a particular function, especially with edge cases.

File: editor/server_api.cpp [hunk 18]
```diff
@@ -0,0 +1,125 @@
...
+  }
+  else
+    LOG(LWARNING, ("CreateNode request has failed:", response.first));
+
+  return false;
+}
+
+bool ServerApi06::ModifyNode(string const & nodeXml, uint64_t nodeId) const
+{
+  OsmOAuth::Response const response = m_auth.Request("/node/" + strings::to_string(nodeId), "PUT", move(nodeXml));
```
Comment (ygorshenin) [inline editor/server_api.cpp]: Ехал move через move, видит move - move - move, сунул move move в move move move move move.

File: editor/server_api.cpp [hunk 19]
```diff
@@ -0,0 +1,125 @@
...
+  string const method = "/user/" + UrlEncode(userName);
+  return m_auth.DirectRequest(method, false).first;
+}
+
+OsmOAuth::Response ServerApi06::GetXmlFeaturesInRect(m2::RectD const & latLonRect) const
+{
+  using strings::to_string_dac;
+
+  // Digits After Comma.
+  static constexpr double const kDAC = 7;
```
Comment (ygorshenin) [inline editor/server_api.cpp]: Either constexpr or const are redundant.

File: editor/ui2oh.cpp [hunk 20]
```diff
@@ -0,0 +1,218 @@
...
+
+vector<TWeekdays> SplitIntoIntervals(editor::ui::TOpeningDays const & days)
+{
+  vector<TWeekdays> result;
+  auto const & noInversionDays = RemoveInversion(days);
+
+  auto previous = *begin(noInversionDays);
+  result.push_back({previous});
+
+  for (auto it = next(begin(noInversionDays)); it != end(noInversionDays); ++it)
```
Comment (ygorshenin) [inline editor/ui2oh.cpp]: ASSERT that noInversionDays is not empty.

File: editor/osm_auth.cpp (hunk 10 above)
Comment (deathbaba) [inline editor/osm_auth.cpp]: @Zverik It's a bug, probably your google auth didn't work because of it.
@ygorshenin good catch, thanks!

File: indexer/feature_algo.cpp [hunk 21]
```diff
@@ -198,4 +198,4 @@ double GetMinDistanceMeters(FeatureType const & ft, m2::PointD const & pt)
   return GetMinDistanceMeters(ft, pt, FeatureType::BEST_GEOMETRY);
 }
 
-}
+} // namespace feature
```
Comment (ygorshenin) [inline indexer/feature_algo.cpp]: two spaces, please.

File: indexer/feature_algo.hpp [hunk 22]
```diff
@@ -13,4 +13,4 @@ m2::PointD GetCenter(FeatureType const & f);
 double GetMinDistanceMeters(FeatureType const & ft, m2::PointD const & pt, int scale);
 double GetMinDistanceMeters(FeatureType const & ft, m2::PointD const & pt);
 
-}
+} // namespace feature
```
Comment (ygorshenin) [inline indexer/feature_algo.hpp]: two spaces, please.

**Comment by deathbaba** [comment]
> search/reverse_geocoder.cpp:
> В GetNearbyFeatureStreets
> Мы пишем
> ```void doit(params, vector<> & result)```
> вместо
> ```
> vector<> doit(params)
> ```

Это принципиальное негласное правило? Почему нельзя использовать код с возвращением вектора?

File: indexer/osm_editor.cpp [hunk 23]
```diff
@@ -0,0 +1,622 @@
...
+#include "std/unordered_set.hpp"
+
+#include "3party/pugixml/src/pugixml.hpp"
+
+using namespace pugi;
+using feature::EGeomType;
+using feature::Metadata;
+using editor::XMLFeature;
+
+constexpr char const * kEditorXMLFileName = "edits.xml";
```
Comment (ygorshenin) [inline indexer/osm_editor.cpp]: Please, put these constants to anonymous namespace.

File: indexer/osm_editor.cpp [hunk 24]
```diff
@@ -0,0 +1,622 @@
...
+  }
+}
+
+Editor::FeatureStatus Editor::GetFeatureStatus(MwmSet::MwmId const & mwmId, uint32_t index) const
+{
+  // Most popular case optimization.
+  if (m_features.empty())
+    return FeatureStatus::Untouched;
+
+  auto const mwmMatched = m_features.find(mwmId);
```
Comment (ygorshenin) [inline indexer/osm_editor.cpp]: s/mwmMatched/matchedMwm/

File: indexer/osm_editor.hpp [hunk 25]
```diff
@@ -0,0 +1,126 @@
...
+
+  static Editor & Instance();
+
+  void SetMwmIdByNameAndVersionFn(TMwmIdByMapNameFn const & fn) { m_mwmIdByMapNameFn = fn; }
+  void SetInvalidateFn(TInvalidateFn const & fn) { m_invalidateFn = fn; }
+  void SetFeatureLoaderFn(TFeatureLoaderFn const & fn) { m_featureLoaderFn = fn; }
+
+  void LoadMapEdits();
+
+  using TFeatureIDFunctor = function<void(FeatureID const &)>;
```
Comment (ygorshenin) [inline indexer/osm_editor.hpp]: Please, move these usings to the top of the public section, near other using declarations.

File: editor/osm_auth.cpp (hunk 13 above)
Comment (mgsergio) [inline editor/osm_auth.cpp]: No, it's not. As we found out constexpr have special meaning for const char *.
And works something like a macro.

File: editor/server_api.cpp (hunk 19 above)
Comment (mgsergio) [inline editor/server_api.cpp]: +1

File: editor/osm_auth.cpp (hunk 13 above)
Comment (ygorshenin) [inline editor/osm_auth.cpp]: Could you please provide a reference or something, where this meaning is described.

File: indexer/osm_editor.cpp [hunk 26]
```diff
@@ -0,0 +1,622 @@
...
+  auto mwmFound = m_features.find(id);
+  if (mwmFound == m_features.end())
+    return;
+
+  // TODO(AlexZ): Check that features are visible at this scale.
+  // Process only new (created) features.
+  for (auto & index : mwmFound->second)
+  {
+    FeatureTypeInfo & ftInfo = index.second;
+    if (ftInfo.m_status == FeatureStatus::Created &&
```
Comment (ygorshenin) [inline indexer/osm_editor.cpp]: What about Untouched or Modified features?

File: indexer/osm_editor.cpp [hunk 27]
```diff
@@ -0,0 +1,622 @@
...
+        // TODO(AlexZ): Synchronize save after edits.
+        // Call Save every time we modify each feature's information.
+        Save(GetEditorFilePath());
+      }
+    }
+    // TODO(AlexZ): Should we call any callback at the end?
+  };
+
+  // Do not run more than one upload thread at a time.
+  static auto future = async(launch::async, lambda, key, secret, tags);
```
Comment (ygorshenin) [inline indexer/osm_editor.cpp]: This lambda captures this, and on shutdown may cause segfaults, if main thread is destroyed before completion of this aux thread.

File: indexer/osm_editor.hpp (hunk 25 above)
Comment (deathbaba) [inline indexer/osm_editor.hpp]: Some typedefs are much easier to read when defined close to the place of their usage. Don't you mind to leave it as is?

File: indexer/osm_editor.cpp (hunk 27 above)
Comment (deathbaba) [inline indexer/osm_editor.cpp]: this is guaranteed to live as long as it can, because it's static variable instance.

File: indexer/osm_editor.cpp (hunk 26 above)
Comment (deathbaba) [inline indexer/osm_editor.cpp]: They are processed separately, by other methods. These ForEach* used to query new features which are absent in our mwm files.

File: editor/osm_auth.hpp (hunk 14 above)
Comment (deathbaba) [inline editor/osm_auth.hpp]: It's a cross-class type, and it's actively used in osm namespace. I propose to leave it as is.

File: editor/osm_auth.cpp (hunk 11 above)
Comment (deathbaba) [inline editor/osm_auth.cpp]: It's hard to test it separately, as it depends on many external parameters. The only way is to move out some pieces of function's code into separate functions and test them. But in this case code integrity is lost.
We can test it indirectly though, @Zverik let's think about good indirect test.

File: indexer/osm_editor.hpp (hunk 25 above)
Comment (ygorshenin) [inline indexer/osm_editor.hpp]: I'm not insisting. If you think that the code is readable enough, leave it as is.

File: editor/osm_auth.cpp (hunk 13 above)
Comment (deathbaba) [inline editor/osm_auth.cpp]: @mgsergio ping.

File: editor/osm_auth.hpp (hunk 14 above)
Comment (ygorshenin) [inline editor/osm_auth.hpp]: If it's a cross-class type, why not to create a struct KeySecret { string m_key; string m_secret; } instead of type alias? You will be able to create a fancy DebugPrint and other stuff then.

File: editor/opening_hours_ui.hpp (hunk 12 above)
Comment (deathbaba) [inline editor/opening_hours_ui.hpp]: This is a cross-class constant, and it's local for inner namespace editor::ui, let's leave it as is.

File: editor/osm_auth.hpp (hunk 14 above)
Comment (deathbaba) [inline editor/osm_auth.hpp]: We initially tried this approach and later changed it to std::pair, to avoid unnecessary dependency for external code in some cases.

File: editor/opening_hours_ui.hpp (hunk 12 above)
Comment (ygorshenin) [inline editor/opening_hours_ui.hpp]: It's not a constant, it's a typedef. And you cannot guarantree that someone will not create a typedef with the same name in the same namespace somewhere.

Comment (deathbaba) [inline editor/opening_hours_ui.hpp]: That ‘someone’ will get compile error, right? And changes his typedef to a different one after that.

Comment (ygorshenin) [inline editor/opening_hours_ui.hpp]: No, he won't. Language standard just claims that this is an undefined behaviour.

Comment (deathbaba) [inline editor/opening_hours_ui.hpp]: Yes, he is.
```
alexz@AlexMac: ~/Downloads$ cat tt.cc 
#include <string>
#include <vector>

namespace test {
using TPair = std::string;
}

namespace test {
using TPair = std::vector<std::string>;
}

int main(int, char**) {
  test::TPair p;
  return p.size();
}
alexz@AlexMac: ~/Downloads$ clang++ -std=c++11 tt.cc
tt.cc:9:7: error: type alias redefinition with different types ('std::vector<std::string>' vs 'std::string' (aka 'basic_string<char, char_traits<char>, allocator<char> >'))
using TPair = std::vector<std::string>;
      ^
tt.cc:5:7: note: previous definition is here
using TPair = std::string;
      ^
1 error generated.
```

Comment (ygorshenin) [inline editor/opening_hours_ui.hpp]: One particular compiler on a particular machine? Are you kiddin me? Please, verify it on all compilers (especially on compilers from the future) and on all machines in the universe.

Comment (ygorshenin) [inline editor/opening_hours_ui.hpp]: BTW, do you check your programs for data races or uninitialized variables in the same way? One run only on your machine?

File: indexer/osm_editor.cpp (hunk 27 above)
Comment (ygorshenin) [inline indexer/osm_editor.cpp]: OK

File: editor/opening_hours_ui.hpp (hunk 12 above)
Comment (deathbaba) [inline editor/opening_hours_ui.hpp]: No, I'm not kidding. I've checked on many online and offline compilers, even on MSVC one. All of them are correctly give compilation errors (and always did it in the past).

File: editor/ui2oh.cpp (hunk 20 above)
Comment (deathbaba) [inline editor/ui2oh.cpp]: Done.

File: editor/opening_hours_ui.hpp (hunk 12 above)
Comment (ygorshenin) [inline editor/opening_hours_ui.hpp]: My point is that you can not check your code on all compilers and machines. If there's a room for undefined behaviour, in accordance with Merphie law something bad is going to happen, soon or later. Why not to prevent it right here and right now, with 100% guarantees? With only a 5-6 LOCS?

File: 3party/opening_hours/opening_hours.cpp [hunk 28]
```diff
@@ -748,4 +751,28 @@ bool OpeningHours::IsValid() const
...
   return m_valid;
 }
+bool OpeningHours::IsTwentyFourHours() const
+{
+  return m_rule.size() == 1 && m_rule[0].IsTwentyFourHours();
+}
+
+bool OpeningHours::HasWeekdaySelector() const
+{
+  return std::any_of(begin(m_rule), end(m_rule), std::mem_fn(&osmoh::RuleSequence::HasWeekdays));
```
Comment (mpimenov) [inline 3party/opening_hours/opening_hours.cpp]: We do not write ```std::```.

Comment (mpimenov) [inline 3party/opening_hours/opening_hours.cpp]: Ah, it's 3 party. Sorry.

File: editor/opening_hours_ui.hpp (hunk 12 above)
Comment (ygorshenin) [inline editor/opening_hours_ui.hpp]: Okay, seems that we will have the same problem even after replace of using to class declaration. OK, leave this as is.

**Comment by deathbaba** [comment]
Fixed commits with Reverse Geocoder to use a new one from new-search branch:
https://github.com/deathbaba/omim/commit/45c13f73f3d5675d176d4a959af5ebf090387ce5
https://github.com/deathbaba/omim/commit/b4f8596155334a37063631a50f65e8b5c2a5c39c
https://github.com/deathbaba/omim/commit/bb199fc0ad987d1c5310f68466491ec129dd07ad

@ygorshenin @vng @mpimenov PTAL & merge please

**Comment by deathbaba** [comment]
Fixed comments to commits with Reverse Geocoder to use a new one from new-search branch:
https://github.com/deathbaba/omim/commit/84f3a6cf89d863feab1f0c36042288f4a82d5f16
https://github.com/deathbaba/omim/commit/444f2191a7a0f772bf6160682eeb4acb47f1de2f
https://github.com/deathbaba/omim/commit/e06da646a37e0a6edf9cb7d37eff1c7f001e08a4

@ygorshenin @vng @mpimenov PTAL & merge please

**Comment by deathbaba** [comment]
@mpimenov @ygorshenin @vng исправил все ваши комментарии. Давайте мержить.

**Comment by ygorshenin** [comment]
LGTM

**Comment by mpimenov** [comment]
LGTM
